agent/engine/docker_task_engine.go

```diff
@@ -597,6 +610,15 @@ func (engine *DockerTaskEngine) createContainer(task *api.Task, container *api.C
 	seelog.Infof("Created container name mapping for task %s - %s -> %s", task, container, containerName)
 	engine.saver.ForceSave()
 
+	// Create metadata directory and file then populate it with common metadata of all containers of this task
```

Comment:
I have added unit tests for the docker_task_engine_test.go for the cases where metadata service is enabled.